### DISCUSSION TITLE
Issue with Neural Network-based particle picking

### DISCUSSION THREAD

**Discussion by doub1esheep** [nextpyp/.github#120] [Q&A] [answered]
Dear nextpyp Development Team,

I am encountering an issue when using the neural network-based picking in nextpyp. I am working with the EMPIAR-10987 dataset and selected ~50 particles in a tomogram for training, everything seemed to go well. However, when I used the trained model (model_last_contrastive.pth) to perform particle picking, no particles were picked.

I am wondering if there might be a particular reason for this outcome. Additionally, I would appreciate access to the trained model used in the tutorial video, so I can perform a comparison test.

**Comment by abartesaghi** [comment, marked as answer]
You probably want to spread your ~50 particles among 2-3 tomograms (ideally with different defocus). When training, make sure you set `Bounding box = 12`, `Threshold for soft/hard positives = 0.5`, and `Max number of particles = 600`.

**Note**: This assumes your tomograms were reconstructed using `Binning factor for reconstruction = 12` as indicated in the practical.

The trained model is included in the tarball used for the workshop which can be downloaded using: 
```
wget https://nextpyp.app/files/data/nextpyp_workshop_data.tbz
```
After decompressing the file, the trained model can be found in: `10987/model_last_contrastive.pth`

**Reply by doub1esheep** [reply]
Thanks a lot for your suggestions! I spread the ~50 particles among 3 tomograms, this made the training model work well, both for EMPIAR-10987 dataset and my own datasets.
I’ve got a quick question: do the parameters in the "training/evaluation (pyp)" panel only apply during the training stage, or do they also affect the evaluation stage? Just trying to figure out if tweaking them during evaluation makes any difference.

**Reply by abartesaghi** [reply]
There are two parameters that affect evaluation: `Max number of particles` and `Threshold for soft/hard positives`. Increasing the former or decreasing the latter should give more particles.
